### PR TITLE
Improved indexer code

### PR DESCRIPTION
**Pull request by Injector-Spenral**
Changed indexer code to comply with new architecture requirements. The majority of the indexer code has been moved to components/indexer.py, and the four methods required by the architecture doc have been added. Also, a third limit switch and motor have been added for the upcoming prototype.

### CODE REVIEW

File: components/indexer.py [hunk 1]
```diff
@@ -5,15 +5,42 @@ class Indexer:
...
-        for motor in self.indexer_motors:
-            motor.stopMotor()
+            for i in range(
+                1, len(motor_states)
+            ):  # Disable motor if switch and previous switch are pressed
+                if not switch_results[i] and not switch_results[i - 1]:
+                    motor_states[i] = False
+
+            for i in range(len(motor_states)):  # Set all motors to required states
+                if motor_states[i]:
```
Comment (auscompgeek) [inline components/indexer.py]: Why are you looping over `range(len(motor_states))` then accessing `motor_states` with that value?

File: components/indexer.py [hunk 2]
```diff
@@ -5,15 +5,42 @@ class Indexer:
...
 
-    def setup(self):
-        pass
+    def execute(self) -> None:
+        if self.indexing:
+            motor_states = [True] * len(self.indexer_motors)
+            switch_results = [switch.get() for switch in self.indexer_switches]
+            if not switch_results[
+                0
+            ]:  # Test the back switch (because all others require 2 switches)
```
Comment (auscompgeek) [inline components/indexer.py]: Why is this if statement written over multiple lines?

File: components/indexer.py [hunk 3]
```diff
@@ -5,15 +5,42 @@ class Indexer:
...
+                motor_states[0] = False
 
-    def on_enable(self):
-        for motor in self.indexer_motors:
-            motor.stopMotor()
+            for i in range(
+                1, len(motor_states)
+            ):  # Disable motor if switch and previous switch are pressed
```
Comment (auscompgeek) [inline components/indexer.py]: Why is this for loop written over multiple lines?

File: components/indexer.py (hunk 1 above)
Comment (Injector-Spenral) [reply components/indexer.py]: Because we then use that value to access indexer_motors.

**Pull request merged**